packages/donor/src/screens/UpcommingDonation/TimeFrameAfterDonationUtil.tsx

```diff
@@ -0,0 +1,34 @@
+import { BookedAppointment, DateUtils } from "@zm-blood-components/common";
+
+export interface ITimeFrameAfterDonationPopupProps {
+  getCurrentDateTime: () => Date;
+  bookedAppointment: BookedAppointment;
+  daysLimit: number; // 1 = 24 hours, 2 = 48 hours, ...
+}
+
+export function TimeFrameAfterDonationPopup({
+  getCurrentDateTime,
+  bookedAppointment,
+  daysLimit,
+}: ITimeFrameAfterDonationPopupProps) {
+  const currentDayTime = getCurrentDateTime();
+  const numberOfDaysBetweenDates = DateUtils.getNumberOfDaysBetweenDates(
+    bookedAppointment.donationStartTimeMillis,
+    currentDayTime.getTime()
+  );
+  const currentDayMillis = currentDayTime.getTime();
+  // Checks if the donation is in the time frame
+  if (!(numberOfDaysBetweenDates <= daysLimit)) {
```

Comment:
Could we make this expression clearer?

Reply:
(numberOfDaysBetweenDates > daysLimit || numberOfDaysBetweenDates  < 0) {
return false
}